core/lib/rom/setup/finalize/finalize_relations.rb

```diff
@@ -57,9 +57,19 @@ def build_relation(klass, registry)
         # TODO: raise a meaningful error here and add spec covering the case
         #       where klass' gateway points to non-existant repo
         gateway = @gateways.fetch(klass.gateway)
+
+        if klass.schema_proc && !klass.schema
+          plugins = schema_plugins
+          resolved_schema = klass.schema_proc.call do
+            plugins.each { |plugin| app_plugin(plugin) }
+          end
+
+          klass.schema(resolved_schema)
```

Comment:
...ahh it's here...can we add `set_schema!(schema)` instead? This could be used in `inherited` hook to where we inherit schema from a parent class.